src/pages/Jobs.js

```diff
@@ -2,27 +2,52 @@ import React, { useState, useEffect } from "react";
 import JobList from "../components/JobCard";
 import { faSearch } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import { Row, Col, Container } from "react-bootstrap";
+import { Row, Col, Container, Form } from "react-bootstrap";
+import { useHistory } from "react-router-dom";
+
 export default function Jobs() {
   let originalList = [];
-  let filteredList = [];
   let [jobs, setJobs] = useState([]);
-  let keyword = "";
+  let [keyword, setKeyword] = useState("");
+  const [encode, setEncode] = useState("");
+  let history = useHistory();
+  let searchTerm = [];
 
   const getData = async () => {
     //let url ="http://localhost:3001/jobs";
     let url = `https://my-json-server.typicode.com/legobitna/Itviec/jobs`;
     let data = await fetch(url);
     let result = await data.json();
     originalList = result;
+    getQuery(originalList);
+  };
 
-    setJobs(result);
+  const getQuery = (arr) => {
+    let filteredList = [];
+    let queryString = window.location.search;
+    let splitQueryString = queryString.split("?q=");
+    if (splitQueryString.length > 1) {
+      searchTerm = splitQueryString[1].split("%20");
+      filteredList = arr.filter((item) =>
+        searchTerm.some((key) =>
```

Comment:
this is debatable, but okay. if you have "the" in a movie title, you'll get every movie with `the`. but it's okay i guess.